validator/src/commands/run/execute.rs

```diff
@@ -192,16 +192,12 @@ pub fn execute(
         .unwrap_or(DEFAULT_TPU_COALESCE);
 
     // Canonicalize ledger path to avoid issues with symlink creation
-    let ledger_path = create_and_canonicalize_directories([&ledger_path])
-        .unwrap_or_else(|err| {
-            eprintln!(
-                "Unable to access ledger path '{}': {err}",
-                ledger_path.display(),
-            );
-            exit(1);
-        })
-        .pop()
-        .unwrap();
+    let ledger_path = create_and_canonicalize_directory(ledger_path).map_err(|err| {
```

Comment:
I changed to `create_and_canonicalize_directory` (this takes singular / old took multiple) to remove some extra burden for the `.pop().unwrap()`